### DISCUSSION TITLE
Tasmota send and receive with knx hardware not receiving

### DISCUSSION THREAD

**Discussion by WoutervanZijl** [ascillato/Tasmota_KNX#586] [Q&A]
Hello Arend,

Currently i have my system running!
I have 15 shelly for switches and I bought some knx hardware. Knx theben for my heating. Thermostat for scenes and temperature. Sv jung server for visualisation of my system. I am very happy with tasmota! And maybe I am not fully using tasmota to make the system stable. 

If I would send a address from the server to a light it wil be send 1 time. Sometimes the light does not respond. Is my telegram then missing or is it gone or should I fix my Wifi or do I need to use more tasmota decices to receive this telegram and send it to the same light but with a adjusted knx telegram to make the system more fails save.

Can I make a rule that after the light is switch on the output would be send again after 1 or 2 minutes. Because sometimes after pressing the button to fast. The light in the app turns on but the light is off.

My access point is a cisco 145ac seperate from the standard internet to put all the devices as a closed circuit. 

If you have any suggestions then I am happy to hear.

Regards,

Wouter van Zijl

**Comment by ascillato** [comment]
Hi,

> Is my telegram then missing or is it gone or should I fix my Wifi or do I need to use more tasmota decices to receive this telegram and send it to the same light but with a adjusted knx telegram to make the system more fails save.

The KNX IP telegrams are being sent as UDP multicast. UDP multicast is not super reliable and depends a lot on the capabilities of the Wi-Fi Router. Many routers discard UDP messages if they are busy (Check if your wifi router has any option as enable multicast or IGMP). Besides that, some routers don't care if a device subscribed to the multicast address is sleeping or not, and they just send the message to the air without anyone listening (Set sleep to 0 in your Tasmota device).

> Can I make a rule that after the light is switch on the output would be send again after 1 or 2 minutes. Because sometimes after pressing the button to fast. The light in the app turns on but the light is off.

Yes, you can make a rule to use the teleperiod message time to send also the status of the light by KNX. There are some examples that can be adapted to that in the docs.

**Reply by WoutervanZijl** [reply]
Thanks! I changed all the tasmota’s to sleep 0.  I have some extra option in my cisco.
Is this maybe helpfull or not. See pictures.

I there is an advantage to turn it on pls let me know. I am no expert in networking. Thats the next chapter someday.
![6D5D755C-71B0-402E-99A7-63B4A6DE3C96](https://user-images.githubusercontent.com/74825412/156525933-c940faa8-5112-47c7-887a-cccc03047af9.jpeg)
![7E512DBC-13E8-40D8-9A8C-8753F586892C](https://user-images.githubusercontent.com/74825412/156525937-549ded4c-744d-4656-8515-ed917fcbfbd0.jpeg)
![1ADDBE9E-A545-496C-B923-5D8E297091C4](https://user-images.githubusercontent.com/74825412/156525938-ee51a411-48f9-4514-b6b4-b4c56ddf1af1.jpeg)

**Reply by ascillato** [reply]
Doesn't seem to have many options related to multicast messages. Try with sleep 0 to see if that improves your setup.

**Reply by WoutervanZijl** [reply]
Yes I wil try this and see how it goes.
Thanks

**Reply by WoutervanZijl** [reply]
![CA2D4871-7EB7-4DB1-93DA-42A314089CDC](https://user-images.githubusercontent.com/74825412/156699247-178d2ccd-cf77-44e2-ad1e-06534cf85104.jpeg)

could this be any help someway?

**Reply by ascillato** [reply]
I really don't know. You should try it. Seems that is for a different thing (media streaming specifics) but you can try if that improves. The options that I mentioned that are in some routers are called ENABLE MULTICAST (YES / NO) and MULTICAST IGMP VERSION (1 to 3 to select). In your router seems to be enabled, otherwise you won't receive any KNX message.
